### PR TITLE
[entropy_src/rtl] Fix observe fifo overflow reg

### PR DESCRIPTION
**Pull request by h-filali**
This PR latches the observe FIFO overflow register signal instead of only pulsing it.
For a more detailed explanation have a look at the individual commit messages.

This PR also contains a commit that changes a mubi4 test true signal from strict to loose.
This is done as a follow up for #21369.

Closes [#21279](https://github.com/lowRISC/opentitan/issues/21279)

### CODE REVIEW

**Comment by h-filali** [comment]
Added @jwnrt and @pamaury as reviewers since this might interest you.

File: hw/ip/entropy_src/data/entropy_src.hjson [hunk 1]
```diff
@@ -1351,7 +1351,7 @@
     },
     { name:     "FW_OV_RD_FIFO_OVERFLOW",
       desc:     "Firmware override observe FIFO overflow status",
-      swaccess: "rw0c",
+      swaccess: "ro",
```
Comment (vogelpi) [inline hw/ip/entropy_src/data/entropy_src.hjson:1354]: This is not a documentation change :-)
Maybe it doesn't make sense to split up this small PR. It's only +16 lines...

Comment (h-filali) [reply hw/ip/entropy_src/data/entropy_src.hjson:1354]: I squashed the first two commits.

**Review by pamaury — APPROVED** [review]
Cannot comment on the RTL but the register/doc and software change look good to me. Thanks @h-filali

**Comment by h-filali** [comment]
@pamaury and @vogelpi thanks both for reviewing. Let's see what the CI tells us and then merge!

**Pull request merged**